src/rtc_pl031.rs

```diff
@@ -0,0 +1,639 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! ARM PL031 Real Time Clock
+//!
+//! This module implements a PL031 Real Time Clock (RTC) that provides a long
+//! time base counter. This is achieved by generating an interrupt signal after
+//! counting for a programmed number of cycles of a real-time clock input.
+//!
+use std::result::Result;
+use std::time::Instant;
+
+use crate::Trigger;
+
+// As you can see in
+//  https://static.docs.arm.com/ddi0224/c/
+//  real_time_clock_pl031_r1p3_technical_reference_manual_DDI0224C.pdf
+//  at section 3.2 Summary of RTC registers, the total size occupied by this
+//  device is 0x000 -> 0xFFC + 4 = 0x1000.
```

Comment:
(Mostly) Blindly copied from Firecracker.  It's referring to the size occupied by the device in bytes (i.e., from byte 0x000 to byte 0xFFF).

You can see it in `AMBA_ID_HIGH`, which sets an upper bound on the register addresses that specify the peripheral and device IDs (4 byte registers between 0xFE0 and 0xFFC).

Technically, only the first 8 bits of the register at 0xFFC are used, but since the register accesses are supposed to be 4 byte aligned, we should assume that the whole memory space between 0x000 and 0xFFF is in use.

That said, I think `AMBA_ID_HIGH` should be set to `0xFFF`, not `0x1000` as the highest addressable byte in the device, and I've made that change.

It's also possible that I've totally misunderstood your question though...

Reply:
Ok. I see. I just didn't understand why the comment is needed. It doesn't look like it refers to anything in the next line :laughing: We can leave it as is.